packages/model-viewer/src/features/scene-graph/three-dom-element.ts

```diff
@@ -32,12 +34,12 @@ type CorrelatedObjects = Set<Object3D>|Set<Material>|Set<Texture>;
 export class ThreeDOMElement {
   readonly[$onUpdate]: () => void;
   // The canonical GLTF or GLTFElement represented by this facade.
-  readonly[$sourceObject]: GLTFElement;
+  readonly[$sourceObject]: GLTFElement|null;
   // The backing Three.js scene graph construct for this element.
   readonly[$correlatedObjects]: CorrelatedObjects|null;
 
   constructor(
-      onUpdate: () => void, element: GLTFElement,
+      onUpdate: () => void, element: GLTFElement|null,
```

Comment:
All the simplifications hinge on this, this seems like an appropriate change given that we want create new objects that will not have a backing glTF file.

Reply:
Correct me if I'm wrong, but it looks to me like the important change you made is making each inheriting constructor take `GLTFElement|null`; as far as I can tell you do in fact pass a real source object to the super call in each one. The one exception perhaps being TextureInfo, but I'm wondering (based on your other comments) if that class simply should not inherit from ThreeDOMElement at all.

Reply:
Actually when you create a texture using the createTexture() API null is provided for all gltf data, which simplifies a lot of things. Elsewhere in the code null is 'allowed' but not directly specified, this makes setting up TextureInfos with empty textures much more simple because you don't have to have branching logic based on whether the gltf data is real or not. But your right TextureInfo doesn't need to be a ThreeDomElement, but that makes me wonder what material difference is ThreeDomElement making at this stage and why have it all, if I can remove something in the middle of the Dom and it makes no difference than.... 🤷  why have it all? I should  look deeper into that existential question.

Reply:
That's a fair point, but let's not go too far down the rabbit hole. I'm suggesting TextureInfo is a special case because as you mentioned, it doesn't really have a correlation to three.js. I think of this base element as representing something that's related to three.js (based on its members).

Reply:
Done.

Reply:
You said done, but this still says `|null`. Am I correct that it's no longer required after your other changes? If so, it may also render some of your console checks unnecessary.

Reply:
Right it's still needed in order to represent image, sampler, and texture (and more to come I think) that _don't_ have glTF data.

Reply:
Went back and looked at this, yeah, I think I can put this back.